misk-aws/src/test/kotlin/misk/jobqueue/sqs/SqsJobQueueServiceTest.kt

```diff
@@ -0,0 +1,92 @@
+package misk.jobqueue.sqs
+
+import com.amazonaws.services.sqs.AmazonSQS
+import com.google.common.util.concurrent.AbstractService
+import com.google.common.util.concurrent.ServiceManager
+import com.google.inject.util.Modules
+import misk.ServiceModule
+import misk.jobqueue.JobConsumer
+import misk.jobqueue.JobQueue
+import misk.jobqueue.QueueName
+import misk.jobqueue.subscribe
+import misk.testing.MiskExternalDependency
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.lang.Thread.sleep
+import java.util.concurrent.LinkedBlockingDeque
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** This is separate from [SqsJobQueueTest] because we don't want the services started automatically. */
+@MiskTest
+internal class SqsJobQueueServiceTest {
+  @MiskExternalDependency private val dockerSqs = DockerSqs
+  @MiskTestModule private val module =
+      Modules.combine(SqsJobQueueTestModule(dockerSqs.credentials, dockerSqs.client), ServiceModule<ManualStartService>())
+  @Inject private lateinit var sqs: AmazonSQS
+  @Inject private lateinit var queue: JobQueue
+  @Inject private lateinit var consumer: JobConsumer
+  @Inject private lateinit var serviceManager: ServiceManager
+  @Inject private lateinit var manualStartService: ManualStartService
+
+  private val queueName = QueueName("sqs_job_queue_service_test")
+
+  @BeforeEach fun createQueues() {
+    sqs.createQueue(queueName.value)
+  }
+
+  @AfterEach
+  internal fun tearDown() {
+    serviceManager.stopAsync()
+    serviceManager.awaitStopped(20, TimeUnit.SECONDS)
+  }
+
+  @Test fun jobsNotHandledUntilAllServicesAreRunning() {
+    val log = LinkedBlockingDeque<String>()
+
+    sleep(100)
```

Comment:
can we avoid having all these actual sleeps in here? It adds unnecessary test runtime. It's also not clear why we need them all, since the job handler shouldn't run until manual start service is started if I understand correctly.

Reply:
Removed the sleeps.

Reply:
Put the sleeps back! It’s the simplest way to make sure that if we regress here we’ll catch it. Using latches or other concurrency mechanisms is possible but it adds complexity.

Reply:
Put them back.

Reply:
What does this catch? If the job was running before the services were ready then the job would process the log.put's?

Reply:
Then the log.puts will be out-of-order and we know we have a problem.

Reply:
Why do the sleeps help here though? For example, nothing happens before this first sleep.